### PR TITLE
Initial SymCrypt + SymCrypt-OpenSSL integration

### PR DESCRIPTION
**Pull request by samuel-lee-msft**
+ Based on Spencer's previous work (#1833), but updated to pull jitterentropy-library separately, and address comments on that PR.
+ Introduces SymCrypt and SymCrypt-OpenSSL packages, and updates the OpenSSL package with a commented out patch to make the SCOSSL engine (provided by SymCrypt-OpenSSL package) used by default by any OpenSSL user.
  

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Introducing SymCrypt and SymCrypt-OpenSSL packages with a patch for OpenSSL is a step towards using SymCrypt as the default FIPS certified module within Mariner in future.

###### Change Log  

- Add SymCrypt package
- Add SymCrypt-OpenSSL package
- Add (not currently enabled by default) patch for OpenSSL to use SCOSSL engine by default

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- local build

### CODE REVIEW

File: SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ The CBL-Mariner SPEC files originated from a variety of sources with varying lic
...
 | OpenMamba | [Openmamba GPLv2 License](https://www.gnu.org/licenses/old-licenses/gpl-2.0.txt) | bash-completion |
 | TreasureData | [ASL 2.0 License](http://www.apache.org/licenses/LICENSE-2.0) | td-agent |
```
Comment (samuel-lee-msft) [inline SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md:9]: Should the jitterentropy-library license also be referenced here? It's not currently being referenced for the OpenSSL patch.

File: SPECS/SymCrypt/SymCrypt.spec [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+# package list!
+
+%files
+%license LICENSE
+%{_libdir}/libsymcrypt.so
+%{_includedir}/*
+
+%changelog
+* Mon Jan 31 2022 Samuel Lee <saml@microsoft.com> - 101.0.0-1
+- Original version for CBL-Mariner
```
Comment (samuel-lee-msft) [inline SPECS/SymCrypt/SymCrypt.spec:66]: I don't understand what the SPEC linter wants from me here - it seems to be indicating that I should be committing with an empty changelog?

File: SPECS/openssl/openssl.spec [hunk 3]
```diff
@@ -43,6 +41,9 @@ Patch20:        openssl-1.1.1-jitterentropy.patch
 Patch21:        openssl-1.1.1-drbg-seed.patch
 Patch22:        CVE-2021-3711.patch
 Patch23:        CVE-2021-3712.patch
+Patch24:        openssl-1.1.1-fips-SymCrypt.patch
+%define __requires_exclude ^/(bin|usr/bin).*$
```
Comment (christopherco) [inline SPECS/openssl/openssl.spec:45]: nit - Apologies that the linter complained about this but please keep these defines at the top of the spec file.

**Review by christopherco — APPROVED** [review]
Thanks for making these changes and overall LGTM. I do see our cgmanifest action is unable to pull the tarball from the github URLs provided. Could you check and make sure we can pull the source tarball from the Symcrypt repo?

File: SPECS/SymCrypt/SymCrypt.spec (hunk 2 above)
Comment (christopherco) [reply SPECS/SymCrypt/SymCrypt.spec]: False positive from the linter. We'll be fixing it in our tool but for now, you can safely ignore it.

**Comment by samuel-lee-msft** [comment]
> Thanks for making these changes and overall LGTM. I do see our cgmanifest action is unable to pull the tarball from the github URLs provided. Could you check and make sure we can pull the source tarball from the Symcrypt repo?

I have just published the required changes to the SymCrypt repo, and the cgmanifest action seems to be happy now.

**Review by christopherco — APPROVED** [review]

File: SPECS/openssl/openssl.spec [hunk 4]
```diff
@@ -133,6 +134,7 @@ cp %{SOURCE4} test/
 %patch19 -p1
 %patch20 -p1
 %patch21 -p1
+# %patch22 -p1
```
Comment (PawelWMS) [inline SPECS/openssl/openssl.spec:137]: Let's either remove the patch or apply it.

**Pull request merged**